tests/mir-opt/async_closure_shims.rs

```diff
@@ -29,10 +29,13 @@ async fn call_once(f: impl AsyncFnOnce(i32)) {
     f(1).await;
 }
 
+async fn call_normal<F: Future<Output = ()>>(f: &impl Fn(i32) -> F) {
+    f(1).await;
+}
+
 // EMIT_MIR async_closure_shims.main-{closure#0}-{closure#0}.coroutine_closure_by_move.0.mir
-// EMIT_MIR async_closure_shims.main-{closure#0}-{closure#0}.coroutine_closure_by_mut.0.mir
```

Comment:
Keep this one around (with the `_ref` rename)

Reply:
There is no by-ref shim for this coroutine, since we're not calling the `Fn` for it. That's why I had to add another closure below that has no captured variables.